test/server/cards/01_SOR/units/ZebOrreliosHeadstrongWarrior.spec.ts

```diff
@@ -0,0 +1,53 @@
+describe('Zeb Orrelios, Headstrong Warrior', function () {
+    integration(function (contextRef) {
+        describe('Zeb Orrelios\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['zeb-orrelios#headstrong-warrior', 'swoop-racer'],
+                        spaceArena: ['green-squadron-awing']
+                    },
+                    player2: {
+                        groundArena: ['superlaser-technician', 'consular-security-force', 'steadfast-battalion'],
+                    }
+                });
+            });
+
+            it('should deal 4 damage to a ground unit when he kill someone and survives', function () {
+                const { context } = contextRef;
+
+                // kill superlaser technician, should deal 4 to a ground unit
+                context.player1.clickCard(context.zebOrrelios);
+                context.player1.clickCard(context.superlaserTechnician);
+                context.player2.clickPrompt('Put Superlaser Technician into play as a resource and ready it');
+                expect(context.player1).toBeAbleToSelectExactly([context.zebOrrelios, context.swoopRacer, context.consularSecurityForce, context.steadfastBattalion]);
+                expect(context.player1).toHaveChooseNoTargetButton();
+                context.player1.clickCard(context.consularSecurityForce);
+                expect(context.consularSecurityForce.damage).toBe(4);
+                expect(context.player2).toBeActivePlayer();
+
+                // reset
+                context.zebOrrelios.exhausted = false;
+                context.setDamage(context.zebOrrelios, 0);
+                context.setDamage(context.consularSecurityForce, 0);
+                context.player2.passAction();
+
+                // consular security force is not defeat, nothing happen
+                context.player1.clickCard(context.zebOrrelios);
+                context.player1.clickCard(context.consularSecurityForce);
+                expect(context.player2).toBeActivePlayer();
```

Comment:
Few more cases it would be good to have since his trigger is kind of complicated:

- Enemy unit attacks him and dies, ability not triggered
- Another friendly unit defeats an enemy unit, ability not triggered

And one additional one that's kind of complicated: Zeb's trigger doesn't say that _he himself_ has to defeat the unit like Mace Windu does, it just says the unit has to _be defeated_. So technically if the defender is defeated by some "on attack" ability, Zeb's ability should still activate.

It's actually perfect that  you also implemented Vambrace Flamethrower in this PR because that is exactly the way to test this. To summarize - let's have a test where Zeb has Flamethrower attached, attacks a unit and defeats it with Flamethrower ability, and then check that Zeb ability still activates. That could maybe be its own `describe` block since it's a pretty different setup.